src/api.cpp

```diff
@@ -0,0 +1,63 @@
+//
+// Created by clemens on 24.06.25.
+//
+
+#include "api.hpp"
+// clang-format off
+#include "ch.h"
+#include "hal.h"
+// clang-format on
+#include <mongoose.h>
+
+#include <filesystem/file.hpp>
+
+static THD_WORKING_AREA(waRestApi, 4096);
+
+// HTTP server event handler function
+static void ev_handler(struct mg_connection *c, int ev, void *ev_data) {
+  if (ev == MG_EV_HTTP_MSG) {
+    struct mg_http_message *hm = (struct mg_http_message *)ev_data;
+    if (mg_match(hm->method, mg_str("POST"), NULL)) {
+      if (mg_match(hm->uri, mg_str("/api/robot_type"), NULL)) {
+        File robot_type;
+        robot_type.open("robot_type.txt");
```

Comment:
As mentioned in Discord, I think it might be more future-proof to use JSON everywhere, for example to have subtypes like Worx with or without knob.